scripts/zkai_attention_kv_d8_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,716 @@
+    path.parent.mkdir(parents=True, exist_ok=True)
+    row = {column: result[column] for column in TSV_COLUMNS}
+    with path.open("w", encoding="utf-8", newline="") as handle:
+        writer = csv.DictWriter(
+            handle, fieldnames=TSV_COLUMNS, delimiter="\t", lineterminator="\n"
+        )
+        writer.writeheader()
+        writer.writerow(row)
+    with path.open("r", encoding="utf-8", newline="") as handle:
+        rows = list(csv.DictReader(handle, delimiter="\t"))
+    if len(rows) != 1 or set(rows[0]) != set(TSV_COLUMNS):
+        raise AttentionKvD8FusedSoftmaxTableGateError("TSV round-trip drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)